pkgs/by-name/no/noctalia-shell/package.nix

```
@@ -0,0 +1,92 @@
+{
+  fetchurl,
+  lib,
+  stdenvNoCC,
+  # build
+  qt6,
+  quickshell,
+  # runtime deps
+  brightnessctl,
+  cava,
+  cliphist,
+  ddcutil,
+  matugen,
+  wlsunset,
+  wl-clipboard,
+  imagemagick,
+  wget,
+  python3,
+  gpu-screen-recorder, # optional
+}:
+let
+  runtimeDeps = [
+    brightnessctl
+    cava
+    cliphist
+    ddcutil
+    matugen
+    wlsunset
+    wl-clipboard
+    imagemagick
+    wget
+    python3
+  ]
+  ++ lib.optionals (stdenvNoCC.hostPlatform.system == "x86_64-linux") [
+    gpu-screen-recorder
+  ];
+in
+stdenvNoCC.mkDerivation rec {
```

```suggestion
stdenvNoCC.mkDerivation (finalAttrs: {
```
Should be preferred nowadays, #315337